### ISSUE TITLE
Add human readable messages for pydantic errors

### ISSUE THREAD

**Issue by Panaetius** [SwissDataScienceCenter/renku-data-services#309]

**Comment by leafty** [comment]
IMO, this is bad design. A better approach is to have error object be part of the API spec. This way, when the UI receives an error, it has all it needs to translate it to a human-readable message, and update the phrasing of the error message without having to impact the backend API. Also, doing this translation at the UI layer means that any i18n would be kept in the UI/interface/presentation layer and not at the backend layer.

**Comment by Panaetius** [comment]
I'm fine with that as well. I originally created this as it was asked for by the UI team.

Though pydantic errors are a bit of a special case, the messages it produces look like 

```
pydantic.error_wrappers.ValidationError: 2 validation errors for UserBase
response -> sent_articles -> 0 -> id
  field required (type=value_error.missing)
response -> sent_articles -> 1 -> id
  field required (type=value_error.missing)
```
which aren't nice to be shown in the UI. But then pydantic already covers these kinds of things https://docs.pydantic.dev/latest/errors/errors/ and we could return `error.json()` as part of the error response or similar.

**Comment by Panaetius** [comment]
yes that's what I mean, just wanted to point out for whoever works on this that pydantic already has facilities for these kinds of things and we don't need to reinvent that.

On a side, we already do some formatting https://github.com/SwissDataScienceCenter/renku-data-services/blob/main/components/renku_data_services/base_api/error_handler.py#L84 so the actual error in the above example would look like `There are errors in the following fields, response.sent_articles.0.id: field required, response.sent_articles.1.id: field required`. Which already is somewhat human readable, though a user won't know what actual thing in the UI those correspond to.
